contracts/src/v0.8/functions/v1_3_0_zksync/ZKSyncFunctionsRouter.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import {FunctionsRouter} from "../v1_0_0/FunctionsRouter.sol";
+import {CallWithExactGasZKSync} from "../../shared/call/CallWithExactGasZKSync.sol";
+
+/**
+ * @title FunctionsRouterZkSync
+ * @notice Specialized version of FunctionsRouter for zkSync that uses
+ *         CallWithExactGasZKSync to control callback gas usage.
+ */
+contract ZKSyncFunctionsRouter is FunctionsRouter {
+  constructor(address linkToken, Config memory config) FunctionsRouter(linkToken, config) {}
+
+  /**
+   * @dev Override the internal callback function to use CallWithExactGasZKSync
+   *      for controlling and measuring gas usage on zkSync.
+   */
+  function _callback(
+    bytes32 requestId,
+    bytes memory response,
+    bytes memory err,
+    uint32 callbackGasLimit,
+    address client
+  ) private override returns (CallbackResult memory) {
+    // 1. Check if client code exists
+    bool destinationNoLongerExists;
+    assembly {
+      destinationNoLongerExists := iszero(extcodesize(client))
```

Comment:
can use `client.code.length == 0`, this assembly isn't required

Reply:
Replaced.